frontend/packages/dev-console/src/components/custom-resource-list/CustomResourceList.tsx

```diff
@@ -42,80 +38,55 @@ const CustomResourceList: React.FC<CustomResourceListProps> = ({
 
       setListItems(newListItems || []);
       setFetched(true);
-
-      if (activeFilters) {
-        const filteredItems = rowFilterReducer(newListItems, activeFilters);
-        setFilteredListItems(filteredItems);
-      } else {
-        setFilteredListItems(newListItems);
-      }
     };
 
     fetchListItems();
 
     return () => {
       ignore = true;
     };
-  }, [dependentResource, fetchCustomResources, queryArg, rowFilters, rowFilterReducer]);
+  }, [dependentResource, fetchCustomResources]);
 
-  const applyRowFilter = React.useCallback(
-    (filter) => {
-      const filteredItems = rowFilterReducer(listItems, filter);
-      setFilteredListItems(filteredItems);
-    },
-    [listItems, rowFilterReducer],
-  );
+  const applyRowFilter = React.useCallback(() => {
+    const queryArgument = queryArg ? getQueryArgument(queryArg) : undefined;
+    const activeFilters = queryArgument?.split(',');
+    const params = new URLSearchParams(window.location.search);
+    const filteredText = params.get(textFilter);
 
-  const applyTextFilter = React.useCallback(
-    (filter) => {
-      const filteredItems = textFilterReducer(listItems, filter);
-      setFilteredListItems(filteredItems);
-    },
-    [listItems, textFilterReducer],
-  );
+    let filteredItems = listItems;
+    if (activeFilters) {
+      filteredItems = rowFilterReducer(filteredItems, activeFilters);
+    }
+    if (filteredText) {
+      filteredItems = textFilterReducer(filteredItems, filteredText);
+    }
+    return filteredItems;
+  }, [listItems, queryArg, rowFilterReducer, textFilter, textFilterReducer]);
 
-  const rowsOfRowFilters = _.map(
-    rowFilters,
-    ({ items: filterItems, reducer, selected, type }, i) => {
-      return (
-        <CheckBoxes
-          key={i}
-          onFilterChange={applyRowFilter}
-          items={filterItems}
-          itemCount={_.size(listItems)}
-          numbers={_.countBy(listItems, reducer)}
-          selected={selected}
-          type={type}
-          reduxIDs={[]}
-        />
-      );
-    },
-  );
+  const filteredListItems = applyRowFilter();
 
   return (
-    <>
-      {textFilterReducer && (
-        <div className="co-m-pane__filter-bar">
-          <div className="co-m-pane__filter-bar-group co-m-pane__filter-bar-group--filter">
-            <TextFilter label="by name" onChange={applyTextFilter} />
-          </div>
-        </div>
-      )}
-
-      <div className="co-m-pane__body">
-        {!_.isEmpty(listItems) && rowsOfRowFilters}
-        <Table
-          data={filteredListItems}
-          defaultSortField={sortBy}
-          defaultSortOrder={sortOrder}
-          aria-label="CustomResources"
-          Header={resourceHeader}
-          Row={resourceRow}
-          loaded={fetched}
-          virtualize
+    <div className="co-m-pane__body">
+      {rowFilters && (
```

Comment:
There can be just `TextFilter` without `RowFilters` in a list page but this assumes that if there is no row filter then we don't need to render the filter toolbar at all.